### ISSUE TITLE
`F.mean_absolute_error` backward fails on ChainerX

### ISSUE THREAD

**Issue by crcrpar** [chainer/chainer#6158]
The backward of `F.mean_absolute_error` fails because chainerx does not support `sign`.

https://github.com/chainer/chainer/blob/f4ba08a521b9154f619b70634b7c5a4f7249944e/chainer/functions/loss/mean_absolute_error.py#L33-L38

* Conditions
```
Platform: Linux-4.4.0-98-generic-x86_64-with-debian-stretch-sid
Chainer: 6.0.0b2
NumPy: 1.15.4
CuPy:
  CuPy Version          : 6.0.0b2
  CUDA Root             : /usr/local/cuda
  CUDA Build Version    : 9020
  CUDA Driver Version   : 9020
  CUDA Runtime Version  : 9020
  cuDNN Build Version   : 7201
  cuDNN Version         : 7201
  NCCL Build Version    : None
iDeep: 2.0.0.post3
```
* Code to reproduce
```python
import chainer, chainerx as chx, chainer.functions as F

x = chx.random.uniform(size=(10, 20)).astype(chx.float32).require_grad()
y = chx.random.uniform(size=(10, 20)).astype(chx.float32).require_grad()
F.mean_absolute_error(x, y).backward()
```
* Error messages, stack traces, or logs
```
In [10]: F.mean_absolute_error(x, y).backward()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-10-f0f80af48979> in <module>
----> 1 F.mean_absolute_error(x, y).backward()

~/.pyenv/versions/3.6.6/envs/chainerdev/lib/python3.6/site-packages/chainer/variable.py in backward(self, retain_grad, enable_double_backprop, loss_scale)
   1333             assert isinstance(arr, chainerx.ndarray)
   1334             chainerx.backward(
-> 1335                 arr, enable_double_backprop=enable_double_backprop)
   1336             return
   1337

~/.pyenv/versions/3.6.6/envs/chainerdev/lib/python3.6/site-packages/chainer/function_node.py in _backward_chainerx(self, target_input_indexes, grad_outputs, retained_inputs, retained_outputs)
    734                     else chainer.Variable(
    735                         gy, requires_grad=gy.is_backprop_required())
--> 736                     for gy in grad_outputs]))
    737
    738         gx_arrs = [gx._data[0] for gx in gxs]

~/.pyenv/versions/3.6.6/envs/chainerdev/lib/python3.6/site-packages/chainer/function_node.py in _backward_target_inputs(self, target_input_indexes, grad_outputs)
    743         # Filters out input gradients that are not required and returns the
    744         # rest.
--> 745         gxs = self.backward(target_input_indexes, grad_outputs)
    746
    747         len_gxs = len(gxs)

~/.pyenv/versions/3.6.6/envs/chainerdev/lib/python3.6/site-packages/chainer/functions/loss/mean_absolute_error.py in backward(self, indexes, grad_outputs)
     35         coeff = gy * gy.data.dtype.type(1. / self.diff.size)
     36         coeff = chainer.functions.broadcast_to(coeff, self.diff.shape)
---> 37         gx0 = coeff * backend.get_array_module(gy.data).sign(self.diff)
     38         return gx0, -gx0
     39

AttributeError: module 'chainerx' has no attribute 'sign'
```

**Comment by niboshi** [comment]
Thank you for reporting!

Actually, currently not all of the Chainer functions are tested against ChainerX.
We assume there are many functions that do not work, just like `F.mean_absolute_error`.
We hope we will soon pull out of this situation by finishing the test rewrites tracked at #6071.

We really appreciate this kind of reports/requests though, because such reports work as indication of demanded functions that should be prioritized. Let's leave this issue open along with #6071.

**Comment by aksub99** [comment]
I'd like to take this up. @niboshi could I be assigned to this?

**Comment by aksub99** [comment]
I have commented under issue #6071

**Comment by crcrpar** [comment]
Seems to have been fixed by #6253, let me close.
